### PR TITLE
created todoApp

### PR DESCRIPTION
**Pull request by vikash-kanauja**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,24 +1,121 @@
-import logo from './logo.svg';
-import './App.css';
+import TodoItem from "./component/TodoItem.js"; 
+import Button from "react-bootstrap/Button"; 
+import { useState } from "react";
```
Comment (amrit56789) [inline src/App.js]: import should be always on top

File: src/App.js [hunk 2]
```diff
@@ -1,24 +1,121 @@
-import logo from './logo.svg';
-import './App.css';
+import TodoItem from "./component/TodoItem.js"; 
+import Button from "react-bootstrap/Button"; 
+import { useState } from "react"; 
 
 function App() {
+  // State variables
+  const [todos, setTodos] = useState([]); // State for storing todo tasks
+  const [todoText, setTodoText] = useState(""); // State for input value
```
Comment (amrit56789) [inline src/App.js]: take a meaning full state variable name.

File: src/App.js [hunk 3]
```diff
@@ -1,24 +1,121 @@
-import logo from './logo.svg';
-import './App.css';
+import TodoItem from "./component/TodoItem.js"; 
+import Button from "react-bootstrap/Button"; 
+import { useState } from "react"; 
 
 function App() {
+  // State variables
+  const [todos, setTodos] = useState([]); // State for storing todo tasks
```
Comment (amrit56789) [inline src/App.js]: todoList

File: src/App.js [hunk 4]
```diff
@@ -1,24 +1,121 @@
...
+  const [todoText, setTodoText] = useState(""); // State for input value
+  const [inputValid, setInputValid] = useState(true); // State for input validation
```
Comment (amrit56789) [inline src/App.js]: here also take meaningful you are showing errors.

File: src/App.js [hunk 5]
```diff
@@ -1,24 +1,121 @@
...
+  const [id, setId] = useState(null); // State for tracking todo item id being updated
```
Comment (amrit56789) [inline src/App.js]: here also.

File: src/App.js [hunk 6]
```diff
@@ -1,24 +1,121 @@
...
+
+  // Function to toggle todo completion status
+  const toggle = (id) => {
```
Comment (amrit56789) [inline src/App.js]: toggle????

Comment (amrit56789) [inline src/App.js]: you can pass whole todo instead of id.

File: src/App.js [hunk 7]
```diff
@@ -1,24 +1,121 @@
...
+
+  // Function to delete a todo
+  const deleteTodo = (id) => {
+    setTodos(todos.filter((data) => data.id !== id));
+    setId(null);
+    setTodoText(""); 
+  };
+
+  // Function to update a todo
+  const updateTodo = (id) => {
```
Comment (amrit56789) [inline src/App.js]: edit Todo.

File: src/App.js [hunk 8]
```diff
@@ -1,24 +1,123 @@
...
-import './App.css';
+import { useState } from "react";
+import TodoItem from "./component/TodoItem.js";
+import Button from "react-bootstrap/Button";
+
 
 function App() {
+  // State variables
+  const [todoList, setTodoList] = useState([]); // State for storing todo tasks
+  const [inputText, setInputText] = useState(""); // State for input value
```
Comment (meanstack9716) [inline src/App.js]: inputText -> todoInputText

File: src/App.js [hunk 9]
```diff
@@ -1,24 +1,123 @@
...
+    const value = e.target.value;
+    setInputText(value);
+    // Validate input (not empty)
+    setInputError(value.trim() !== "");
+  };
+
+  // Function to handle form submission
+  const handleSubmit = (e) => {
+    // Validate input on submit
+    if (inputText.trim() === "") {
```
Comment (meanstack9716) [inline src/App.js]: button is disable in this condition so no need for this

File: package.json [hunk 10]
```diff
@@ -10,6 +10,8 @@
     "react": "^18.2.0",
     "react-bootstrap": "^2.10.2",
     "react-dom": "^18.2.0",
+    "react-icon": "^1.0.0",
```
Comment (meanstack9716) [inline package.json]: remove this

File: src/App.js [hunk 11]
```diff
@@ -1,24 +1,120 @@
-import logo from './logo.svg';
-import './App.css';
+import { useState } from "react";
+import TodoItem from "./component/TodoItem.js";
+import Button from "react-bootstrap/Button";
```
Comment (meanstack9716) [inline src/App.js:2]: always follow best practices libraries should import first then components

File: src/App.js [hunk 12]
```diff
@@ -1,13 +1,17 @@
...
+import DeleteConfirmation from "./component/DeleteConfirmationPopup.js";
+// import DeleteConfirmation from "./component/DeleteConfirmationModal.js";
 function App() {
   // State variables
   const [todoList, setTodoList] = useState([]); // State for storing todo tasks
   const [todoInputText, setTodoInputText] = useState(""); // State for input value
   const [inputError, setInputError] = useState(true); // State for input validation
   const [editTodoId, setEditTodoId] = useState(null); // State for tracking todo item id being updated
+  const [displayConfirmationModal, setDisplayConfirmationModal] = useState(false);
+  const [deleteMessage, setDeleteMessage] = useState(null);
```
Comment (meanstack9716) [inline src/App.js]: This should be pass as a prop

File: src/App.js [hunk 13]
```diff
@@ -80,11 +97,11 @@ function App() {
         {todoList.map((task, index) => (
           <TodoItem
             key={index}
-            id={task.id}
             task={task}
             showOrHideCompleteBadge={showOrHideCompleteBadge}
-            deleteTodo={deleteTodo}
+            // deleteTodo={deleteTodo}
```
Comment (meanstack9716) [inline src/App.js]: remove comments

**Pull request merged**